### ISSUE TITLE
Documentation Clarification - `createMiddleware` and `context`

### ISSUE THREAD

**Issue by jtmcdole** [dart-lang/shelf#411]
`Request` and `Response` are immutable and so is the `context` member.  The documentation for context makes it seem easy to signal information to middleware and handlers.

1. Update `context` to be clear that modifying it on a request or response requires calling `.change()` on those objects
2. Adding a note in createMiddleware about the inability to forward a new request/response

Happy to work on this.  Would also be happy to have a way to easily pass information down a pipeline without manipulating the request object.

**Comment by kevmoo** [comment]
Go for it!

**Comment by kroikie** [comment]
Hi @kevmoo, just to be clear: is it the intension that `createMiddleware` not allow the forwarding of a "changed" request?

I think I made the same assumption as @jtmcdole that `createMiddleware` would allow me to update the request, especially the context or request headers before it is passed on to the next middleware or handler.

Thanks for any clarification.

**Comment by kevmoo** [comment]
Yes! You can use `createMiddleware` to update a request and/or a response!
